### PR TITLE
[style] Fully expose STAT table with fallback

### LINKED ISSUE #1337: Exposing OS/2 table?
I’m working on a small project using HarfBuzz and I need to access a couple of fields from `OS/2` table. It is a simple table and I can just read it in my code, but HarfBuzz already reads it for the font extents so I thought to ask if exposing the rest of the table is something HarfBuzz might want to do?

### PR DESCRIPTION
**Pull request by ebraminio**
Closes #1337
Not working yet (apparently accelerator won't work if table doesn't exist?) uploaded separately to see if the approach seems better.

Maybe the API should memcpy the infos? (iteration)

### CODE REVIEW

File: src/hb-ot-style.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * Since: REPLACEME
+ */
+typedef struct
+{
+  hb_tag_t		tag;
+  hb_ot_name_id_t	axis_name_id;
+  unsigned int		axis_ordering;
+
+  unsigned int		axis_index;
+  unsigned int		value_index;
```
Comment (ebraminio) [inline src/hb-ot-style.h]: specifically for format4.

**Comment by ebraminio** [comment]
>  Not working yet (apparently accelerator won't work if table doesn't exist?) uploaded separately to see if the approach seems better.

Working, but not on empty faces, which I don't know is important or not.

**Comment by ebraminio** [comment]
Two things to consider if general approach looks good,
* Expose elidedFallbackNameID to make this a "full expose"
* Separate API just for axes, deciding if StatAxisRecord.{nameID,ordering} should go there instead.
* Considering designAxisSize also (which is not clear to me how)

File: src/hb-machinery.hh [hunk 2]
```diff
@@ -114,6 +114,7 @@ static inline Type& StructAfter(TObject &X)
 
 #define DEFINE_SIZE_UNBOUNDED(size) \
   DEFINE_INSTANCE_ASSERTION (sizeof (*this) >= (size)) \
+  enum { null_size = (size) }; \
```
Comment (behdad) [inline src/hb-machinery.hh]: Dude!  There's a reason this line is not there!

**Comment by behdad** [comment]
Not going to review this for a while, until other work settles.  Please don't merge any API changes, or machinery or other part changes.  Just pushing tables in without changing common machineries is fine.

File: src/hb-machinery.hh (hunk 2 above)
Comment (ebraminio) [reply src/hb-machinery.hh]: I know... the intention was just to have something to start with as you can see from other added comments also

**Comment by ebraminio** [comment]
Superseded by #1890

**Pull request closed without merging**